Python/PRP/GUI/pyGUIDynDisplayCtrl.cpp

```diff
@@ -0,0 +1,200 @@
+    "Remove all layers from the dynamic display control")
+{
+    self->fThis->clearLayers();
+    Py_RETURN_NONE;
+}
+
+PY_METHOD_VA(GUIDynDisplayCtrl, addMaterial,
+    "Params: animation key\n"
+    "Add a material to the dynamic display control")
+{
+    pyKey* key;
+    if (!PyArg_ParseTuple(args, "O", &key)) {
+        PyErr_SetString(PyExc_TypeError, "addMaterial expects a plKey");
+        return NULL;
+    }
+    if (!pyKey_Check((PyObject*)key)) {
+        PyErr_SetString(PyExc_TypeError, "addMaterial expects a plKey");
+        return NULL;
+    }
+    self->fThis->addMaterial(*key->fThis);
+    Py_RETURN_NONE;
+}
+
+PY_METHOD_VA(GUIDynDisplayCtrl, delMaterial,
+    "Params: idx\n"
+    "Remove a material from the dynamic display control")
+{
+    int idx;
+    if (!PyArg_ParseTuple(args, "i", &idx)) {
+        PyErr_SetString(PyExc_TypeError, "delMaterial expects an int");
+        return NULL;
+    }
+    self->fThis->delMaterial(idx);
+    Py_RETURN_NONE;
+}
+
+PY_METHOD_NOARGS(GUIDynDisplayCtrl, clearMaterials,
+    "Remove all materials from the dynamic display control")
+{
+    self->fThis->clearMaterials();
+    Py_RETURN_NONE;
+}
+
+static PyMethodDef pyGUIDynDisplayCtrl_Methods[] = {
+    pyGUIDynDisplayCtrl_addTextMap_method,
+    pyGUIDynDisplayCtrl_delTextMap_method,
+    pyGUIDynDisplayCtrl_clearTextMaps_method,
+    pyGUIDynDisplayCtrl_addLayer_method,
+    pyGUIDynDisplayCtrl_delLayer_method,
+    pyGUIDynDisplayCtrl_clearLayers_method,
+    pyGUIDynDisplayCtrl_addMaterial_method,
+    pyGUIDynDisplayCtrl_delMaterial_method,
+    pyGUIDynDisplayCtrl_clearMaterials_method,
+    PY_METHOD_TERMINATOR
+};
+
+PY_GETSET_GETTER_DECL(GUIDynDisplayCtrl, textMaps) {
+    PyObject* list = PyTuple_New(self->fThis->getTextMaps().size());
+    for (size_t i = 0; i<self->fThis->getTextMaps().size(); i++)
+        PyTuple_SET_ITEM(list, i, pyPlasma_convert(self->fThis->getTextMaps()[i]));
+    return list;
+}
+
+PY_GETSET_GETTER_DECL(GUIDynDisplayCtrl, layers) {
+    PyObject* list = PyTuple_New(self->fThis->getLayers().size());
+    for (size_t i = 0; i<self->fThis->getLayers().size(); i++)
+        PyTuple_SET_ITEM(list, i, pyPlasma_convert(self->fThis->getLayers()[i]));
+    return list;
+}
+
+PY_GETSET_GETTER_DECL(GUIDynDisplayCtrl, materials) {
+    PyObject* list = PyTuple_New(self->fThis->getMaterials().size());
+    for (size_t i = 0; i<self->fThis->getMaterials().size(); i++)
+        PyTuple_SET_ITEM(list, i, pyPlasma_convert(self->fThis->getMaterials()[i]));
+    return list;
+}
+
+PY_PROPERTY_SETTER_MSG(GUIDynDisplayCtrl, textMaps, "To add textMaps, use addTextMaps()")
+PY_PROPERTY_GETSET_DECL(GUIDynDisplayCtrl, textMaps)
+PY_PROPERTY_SETTER_MSG(GUIDynDisplayCtrl, layers, "To add layers, use addLayer()")
+PY_PROPERTY_GETSET_DECL(GUIDynDisplayCtrl, layers)
+PY_PROPERTY_SETTER_MSG(GUIDynDisplayCtrl, materials, "To add materials, use addMaterial()")
+PY_PROPERTY_GETSET_DECL(GUIDynDisplayCtrl, materials)
```

Comment:
In the refactoring, I was generally moving the SETTER_MSG and GETSET_DECL next to the GETTER_DECL, so all three parts defining a single property would be grouped together.